src/mugs.js

```diff
@@ -1,3 +1,37 @@
+/**
+ * A mug is a question, containing data, bind, and control elements.
+ *
+ * Main Properties
+ *  nodeID
+ *  label
```

Comment:
I think that makes sense for practical reasons, but I'm a little torn.

I feel like we've been moving in a direction where we're defining better interfaces and protocols around mug properties and various plugin extension points. Labels (itext vs default) are one place where I feel like the interface is not quite there yet, but it's close. If we stop trying to keep javaRosa separate it may become even more tightly coupled with the rest of vellum, which is not ideal because it's a big chunk of code to digest, and having that be part of the core means there is that much more that needs to be understood to get the gist of how vellum works.

On the other hand, the `label` and `hintLabel` properties are not well tested, and I'm not even sure it's possible to run vellum with the javaRosa plugin disabled to fully test them. While that seems like an interesting goal, if for no other reason than having smaller XML snippets (no itext, etc.) for testing features that are orthogonal to javaRosa, I'm not sure it's practical.

Unless I've convinced either of you to come to my side (keep javaRosa as a plugin, separate from vellum core), I think I've been outvoted.